doc/design/software-bridge-management.md

```diff
@@ -0,0 +1,508 @@
+	Uplink map[string]string `json:"uplink,omitempty"` // TODO clarify required settings
+}
+
+// BridgeConfig contains some options for linux bridge
+type BridgeConfig struct {
+	VlanFiltering bool `json:"vlan_filtering,omitempty"`
+	// +kubebuilder:validation:Enum=802.1Q;802.1ad
+	VlanProtocol string `json:"vlan_protocol,omitempty"`
+}
+```
+
+_**Note 1**: multiple policies can match single PF (vf range use-case), bridge settings from the policy with
+ the higher priority (which is the one with the lowest `.spec.priority` value) will be applied to PF._
+
+_**Note 2**: multiple NICs can match the same policy on a host. In this case a separate bridge will be created for each NIC._
+
+#### SriovNetworkNodeState CRD
+
+```golang
+
+type SriovNetworkNodeStateSpec struct {
+	DpConfigVersion string     `json:"dpConfigVersion,omitempty"`
+	Interfaces      Interfaces `json:"interfaces,omitempty"`
+	Bridges         Bridges    `json:"bridges,omitempty"`
+}
+
+// SriovNetworkNodeStateStatus defines the observed state of SriovNetworkNodeState
+type SriovNetworkNodeStateStatus struct {
+	Interfaces    InterfaceExts `json:"interfaces,omitempty"`
+	Bridges       Bridges       `json:"bridges,omitempty"`
+	SyncStatus    string        `json:"syncStatus,omitempty"`
+	LastSyncError string        `json:"lastSyncError,omitempty"`
+}
+
+// Bridges contains list of bridges
+type Bridges struct {
+	OVS   []OVSConfigExt         `json:"ovs,omitempty"`
+	Linux []LinuxBridgeConfigExt `json:"linux,omitempty"`
+}
+
+type OVSConfigExt struct {
+	// name of the bridge
+	Name    string               `json:"name"`
+	// bridge-level configuration for the bridge
+	Bridge  OVSBridgeConfig      `json:"bridge,omitempty"`
+	// uplink-level bridge configuration for each uplink(PF).
+	// in the initial implementation will always contain one element
+	Uplinks []OVSUplinkConfigExt `json:"uplinks,omitempty"`
+}
+
+type OVSUplinkConfigExt struct {
+	// pci address of the PF
+	PciAddress string             `json:"pciAddress"`
+	// name of the PF interface
+	Name       string             `json:"name,omitempty"`
+	// configuration from the Interface OVS table for the PF
+	Interface  OVSInterfaceConfig `json:"interface,omitempty"`
+}
+
+type LinuxBridgeConfigExt struct {
+	Name    string                       `json:"name"`
+	Bridge  BridgeConfig                 `json:"bridge,omitempty"`
+	Uplinks []LinuxBridgeUPlinkConfigExt `json:"uplinks,omitempty"`
+}
+
+type LinuxBridgeUPlinkConfigExt struct {
+	PciAddress string             `json:"pciAddress"`
+	Name       string             `json:"name,omitempty"`
+	Interface  OVSInterfaceConfig `json:"interface,omitempty"`
+}
+
+type VirtualFunction struct {
+	// ...existing fields...
+	// contains VF representor name for NICs in switchdev mode
+	RepresentorName string `json:"representorName,omitempty"`
+}
+
+```
+`SriovNetworkNodeState.spec` and `SriovNetworkNodeState.status` should be extended to contain same `Bridges`.
```

Comment:
is this statement still correct?

Reply:
Yes, it is correct. The statement differs from the original one - now is says that we use the same struct in "spec" and "status" of the SriovNetworkNodeState. Also I added note which says that we may want to extend/replace the struct in the future.